src/multiplexer/EventHandler_cgi.cpp

```diff
@@ -3,54 +3,72 @@
 /*                                                        :::      ::::::::   */
 /*   EventHandler_cgi.cpp                               :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: bthomas <bthomas@student.42.fr>            +#+  +:+       +#+        */
+/*   By: okoca <okoca@student.42.fr>                +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/09/28 12:35:44 by bthomas           #+#    #+#             */
-/*   Updated: 2024/10/02 13:49:57 by bthomas          ###   ########.fr       */
+/*   Updated: 2024/10/03 17:06:45 by okoca            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "EventHandler.hpp"
 #include "string.h"
+#include <cstdlib>
+#include <vector>
 
 static void setPipe(int *fd, int end) {
 	if (dup2(fd[end], end) == -1 || close(fd[0]) == -1 || close(fd[1]) == -1) {
 		std::cerr << "Error: could not open pipe for cgi output\n";
-		exit(1);
+		throw (1);
 	}
 }
 
 // forks process - child execve's script
-void EventHandler::startCGI(int clientFd, std::string fname) {
+// return true if success -> false if failed
+bool EventHandler::startCGI(int clientFd, std::vector<std::string> arguments) {
 	pid_t pid;
 	int fd[2];
 
 	if (pipe(fd) == -1) {
 		std::cerr << "Error: could not establish pipe for CGI output\n";
-		return ;
+		return false;
 	}
 	pid = fork();
 	if (pid == -1) {
 		std::cerr << "Error: could not fork process in CGI output\n";
 		close(fd[0]);
 		close(fd[1]);
-		return ;
+		return false;
 	}
 	if (pid == 0) {
 		setPipe(fd, STDOUT_FILENO);
-		char **args = new char*;
-		args[0] = const_cast<char*>(fname.c_str());
-		execve(fname.c_str(),args, environ);
-		delete args;
+		char **args = new char*[3];
+		args[0] = const_cast<char *>(arguments[0].c_str());
+		args[1] = const_cast<char *>(arguments[1].c_str());
```

Comment:
might be good to thing of a way around the const_cast later on. It's a little hacky